lib/IRGen/GenType.cpp

```diff
@@ -423,6 +423,19 @@ static llvm::Value *computeExtraTagBytes(IRGenFunction &IGF, IRBuilder &Builder,
   return phi;
 }
 
+static Address alignAddress(IRGenFunction &IGF,
+		            Address originAddress,
+			    llvm::Value *shift){
+  auto &Builder = IGF.Builder;
+  auto &IGM = IGF.IGM;
+  auto *alignedAddress = Builder.CreateBitCast(originAddress, IGM.Int8PtrTy).getAddress();
+  if (IGM.Triple.getOS() == llvm::Triple::Linux && !IGM.Triple.isLittleEndian()) {
+    alignedAddress = Builder.CreateConstGEP1_32(alignedAddress, 4);
+    alignedAddress = Builder.CreateGEP(alignedAddress, Builder.CreateNeg(shift));
+  }
+  return Address(alignedAddress, Alignment(0));
```

Comment:
Resetting the alignment like this is not correct.  You need to reuse the alignment of the address.